Assets/MRTK/Core/Inspectors/Setup/MixedRealityProjectConfiguratorWindow.cs

```diff
@@ -78,75 +108,514 @@ public static void ShowWindow()
+                case ConfigurationStage.InstallMSOpenXR:
+                    RenderEnableMicrosoftOpenXRPlugin();
+                    break;
+                case ConfigurationStage.InstallBuiltinPlugin:
+                    RenderEnableXRSDKBuiltinPlugin();
+                    break;
+                case ConfigurationStage.ProjectConfiguration:
+                    RenderProjectConfigurations();
+                    break;
+                case ConfigurationStage.ImportTMP:
+                    RenderImportTMP();
+                    break;
+                case ConfigurationStage.ShowExamples:
+                    RenderShowUPMExamples();
+                    break;
+                case ConfigurationStage.Done:
+                    RenderConfigurationCompleted();
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        private void RenderXRPipelineSelection()
         {
-            Instance = this;
+            if (!XRSettingsUtilities.XREnabled)
+            {
+                RenderNoPipeline();
+            }
+            else if (XRSettingsUtilities.LegacyXREnabled)
+            {
+                RenderLegacyXRPipelineDetected();
+            }
+            else
+            {
+                if (XRSettingsUtilities.MicrosoftOpenXREnabled)
+                {
+                    RenderMicrosoftOpenXRPipelineDetected();
+                }
+                else if (XRSettingsUtilities.OpenXREnabled)
+                {
+                    RenderOpenXRPipelineDetected();
+                }
+                else
+                {
+                    RenderXRSDKBuiltinPluginPipelineDetected();
+                }
+            }
+        }
 
-            CompilationPipeline.assemblyCompilationStarted += CompilationPipeline_assemblyCompilationStarted;
-            MixedRealityProjectConfigurator.SelectedSpatializer = SpatializerUtilities.CurrentSpatializer;
+        private void RenderNoPipeline()
+        {
+            if (!XRSettingsUtilities.LegacyXRAvailable)
+            {
+#if UNITY_2020_2_OR_NEWER
+                currentStage = ConfigurationStage.SelectXRSDKPlugin;
+#else
+                currentStage = ConfigurationStage.InstallBuiltinPlugin;
+#endif // UNITY_2020_2_OR_NEWER
+
+                Repaint();
+            }
+            EditorGUILayout.LabelField("XR Pipeline Setting", EditorStyles.boldLabel);
+            EditorGUILayout.LabelField("To build applications for AR/VR headsets you need to specify an XR pipeline. "
+#if UNITY_2019_3_OR_NEWER
+                + $"Unity currently provides the following pipelines in this version ({Application.unityVersion}). Please choose the one you would like to use:"
+                + $"\n\nLegacy XR (Recommended) - Choose this if you want maximum stability and are willing to spend more effort when upgrading the project to Unity 2020. Supports HoloLens, Windows Mixed Reality/OpenVR/Oculus headset."
```

Comment:
I think we can leave Oculus out, since the main MRTK itself doesn't explicitly have support on legacy XR.

```suggestion
                + $"\n\nLegacy XR (Recommended) - Choose this if you want maximum stability and are willing to spend more effort when upgrading the project to Unity 2020. Supports HoloLens and Windows Mixed Reality/OpenVR headsets."
```